Formula/bup.rb

```diff
@@ -17,11 +17,11 @@ class Bup < Formula
 
   depends_on "pandoc" => :build
   depends_on "pkg-config" => :build
-  depends_on "python@3.10"
+  depends_on "python@3.11"
 
   def install
-    ENV["PYTHON"] = Formula["python@3.10"].opt_bin/"python3.10"
-
+    python3 = "python3.11"
+    ENV["BUP_PYTHON_CONFIG"] = "#{python3}-config"
```

Comment:
`PYTHON` is no longer used in 0.33: https://github.com/bup/bup/blob/master/note/0.33-from-0.32.md#notable-changes-in-033-since-032

Build script does detect correct Python without `BUP_PYTHON_CONFIG` due to adding libexec dir, but can set variable to make sure correct one is picked
- With variable:
    ```
    found: python-config (/usr/local/opt/python@3.11/bin/python3.11-config)
    ```
- Without variable:
    ```
    checking for python3.10-config ()
    checking for python3.9-config ()
    checking for python3.8-config ()
    checking for python3.7-config ()
    checking for python3.6-config ()
    checking for python3-config (/usr/local/opt/python@3.11/libexec/bin/python3-config)
    ...
    
    found: python-config (/usr/local/opt/python@3.11/libexec/bin/python3-config)
    ```